<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from mh180_120.urdf.xacro           | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="motoman_mh_180_120">
  <!--plugin for webots_ros2 control-->
  <webots>
    <plugin type="webots_ros2_control::Ros2Control"/>
  </webots>
  <ros2_control name="WebotsControl" type="system">
    <hardware>
      <!-- By default, set up controllers for simulation. This won't work on real hardware -->
      <!-- <plugin>mock_components/GenericSystem</plugin> -->
      <plugin>webots_ros2_control::Ros2ControlSystem</plugin>
    </hardware>
    <joint name="group_1/joint_1">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
    </joint>
    <joint name="group_1/joint_2">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
    </joint>
    <joint name="group_1/joint_3">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
    </joint>
    <joint name="group_1/joint_4">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
    </joint>
    <joint name="group_1/joint_5">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
    </joint>
    <joint name="group_1/joint_6">
      <command_interface name="position"/>
      <command_interface name="velocity"/>
      <state_interface name="position">
        <param name="initial_value">0.0</param>
      </state_interface>
      <state_interface name="velocity"/>
    </joint>
  </ros2_control>
  <!-- link list -->
  <link name="base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_robotcrtl/meshes/mh180/visual/base_link.stl"/>
      </geometry>
      <material name="blue">
        <color rgba="0 0 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_robotcrtl/meshes/mh180/collision/base_link.stl"/>
        <!--box size="0.9 0.5 0.32"/-->
      </geometry>
      <material name="yellow">
        <color rgba="0 1 1 1"/>
      </material>
    </collision>
  </link>
  <link name="link_s">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_robotcrtl/meshes/mh180/visual/link_s.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_robotcrtl/meshes/mh180/collision/link_s.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="link_l">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_robotcrtl/meshes/mh180/visual/link_l.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_robotcrtl/meshes/mh180/collision/link_l.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="link_u">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_robotcrtl/meshes/mh180_120/visual/link_u.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_robotcrtl/meshes/mh180_120/collision/link_u.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="link_r">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_robotcrtl/meshes/mh180_120/visual/link_r.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_robotcrtl/meshes/mh180_120/collision/link_r.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="link_b">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_robotcrtl/meshes/mh180/visual/link_b.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_robotcrtl/meshes/mh180/collision/link_b.stl"/>
      </geometry>
      <material name="yellow"/>
    </collision>
  </link>
  <link name="link_t">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_robotcrtl/meshes/mh180/visual/link_t.stl"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://motoman_robotcrtl/meshes/mh180/collision/link_t.stl"/>
      </geometry>
      <material name="black"/>
    </collision>
  </link>
  <link name="tool0"/>
  <!-- end of link list -->
  <!-- joint list -->
  <joint name="group_1/joint_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.650"/>
    <parent link="base_link"/>
    <child link="link_s"/>
    <axis xyz="0 0 1"/>
    <limit effort="10000" lower="-3.141592653589793" upper="3.141592653589793" velocity="2.181661564992912"/>
  </joint>
  <joint name="group_1/joint_2" type="revolute">
    <origin rpy="0 0 0" xyz="0.325 0 0"/>
    <parent link="link_s"/>
    <child link="link_l"/>
    <axis xyz="0 1 0"/>
    <limit effort="10000" lower="-1.0471975511965976" upper="1.3264502315156905" velocity="2.007128639793479"/>
  </joint>
  <joint name="group_1/joint_3" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 1.150"/>
    <parent link="link_l"/>
    <child link="link_u"/>
    <axis xyz="0 -1 0"/>
    <limit effort="5000" lower="-1.4835298641951802" upper="1.5707963267948966" velocity="2.181661564992912"/>
  </joint>
  <joint name="group_1/joint_4" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.3"/>
    <parent link="link_u"/>
    <child link="link_r"/>
    <axis xyz="-1 0 0"/>
    <limit effort="5000" lower="-6.283185307179586" upper="6.283185307179586" velocity="3.1764992386296798"/>
  </joint>
  <joint name="group_1/joint_5" type="revolute">
    <origin rpy="0 0 0" xyz="1.590 0 0"/>
    <parent link="link_r"/>
    <child link="link_b"/>
    <axis xyz="0 -1 0"/>
    <limit effort="2500" lower="-2.2689280275926285" upper="2.2689280275926285" velocity="3.0543261909900767"/>
  </joint>
  <joint name="group_1/joint_6" type="revolute">
    <origin rpy="0 0 0" xyz="0.225 0 0"/>
    <parent link="link_b"/>
    <child link="link_t"/>
    <axis xyz="-1 0 0"/>
    <limit effort="2500" lower="-6.283185307179586" upper="6.283185307179586" velocity="4.625122517784973"/>
  </joint>
  <joint name="group_1/joint_6-tool0" type="fixed">
    <origin rpy="0 1.5707963267948966 0" xyz="0 0 0"/>
    <parent link="link_t"/>
    <child link="tool0"/>
  </joint>
  <!-- end of joint list -->
  <!-- ROS base_link to Robot Manufacturer World Coordinates transform -->
  <!-- <link name="${prefix}base"/>
    <joint name="${prefix}base_link-base" type="fixed">
      <origin xyz="0 0 0.650" rpy="0 0 0"/>
      <parent link="${prefix}base_link"/>
      <child link="${prefix}base"/>
    </joint> -->
</robot>
